src/dataMap.js

```diff
@@ -413,21 +413,26 @@ class DataMap {
    * @returns {boolean} Returns `false` when action was cancelled, otherwise `true`.
    */
   removeRow(index, amount = 1, source) {
-    let rowIndex = typeof index !== 'number' ? -amount : index;
-    const rowsAmount = this.instance.runHooks('modifyRemovedAmount', amount, rowIndex);
+    let rowIndex = Number.isInteger(index) ? index : -amount; // -amount = taking indexes from the end.
+    const physicalIndexes = this.visualRowsToPhysical(rowIndex, amount);
+    // It handle also callback from the `NestedRows` plugin. Removing parent node has effect in removing children nodes.
+    const removedPhysicalIndexes = this.instance.runHooks('modifyRemovedRows', physicalIndexes);
```

Comment:
it's worth to add a hook to documentation (`pluginHooks.js`).

Reply:
@wojciechczerniak, the old `modifyRemovedAmount` was quite private. New, `modifyRemovedRows` seems to be more useful.

There is even a test for extra case: https://github.com/handsontable/handsontable/pull/6907/commits/cd3b38c646ece46ee84ed66b4d1e6e93bfb448a4, so for me, the situation isn't similar to this introduced within [PR discussion](https://github.com/handsontable/handsontable/pull/6889#discussion_r426590991). I would add the hook and describe it in the release notes. WDYT?

Reply:
I suppose that it should be added after watching the mentioned discussion. Changed within: https://github.com/handsontable/handsontable/pull/6907/commits/650aa1824a666cea705184c5e4fe1d62c96349fa.

Reply:
@aninde Please add information to the release notes :)

Reply:
@scarletfog Please add information to the release notes :)

Reply:
@wszymanski can we say it's a new feature?

Reply:
I think so.

Reply:
ok, thank you! 😄